cosmwasm/ibc-union/app/ucs03-zkgm/src/contract.rs

```diff
@@ -1502,6 +1564,277 @@ fn execute_fungible_asset_order(
         )?))
 }
 
+#[allow(clippy::too_many_arguments)]
+fn execute_stake(
+    deps: DepsMut,
+    env: Env,
+    packet: Packet,
+    stake: Stake,
+    intent: bool,
+) -> Result<Response, ContractError> {
+    // Market makers not allowed to fill staking requests.
+    if intent {
+        return Ok(Response::new().add_message(wasm_execute(
+            env.contract.address,
+            &ExecuteMsg::InternalWriteAck {
+                ack: ACK_ERR_ONLY_MAKER.into(),
```

Comment:
isn't the error message actually the opposite of the error here? i.e. ErrNeverMaker or something

Reply:
it's misleading i'll probably rename the constant because it just means we want to revert the entire transaction without writting a failure ack